XavierGuertin/WildfireVisualizationProject
# #37 - feat(docker): Setting Up Docker

# Docker Setup PR

This PR has been created to set up **Docker** to containerize the **frontend, backend, and database** services for the project.

---

## **Summary**
This PR introduces the following changes:
- Added Dockerfiles for the **frontend** (React/Next.js) and **backend** (Spring Boot).
- Configured `docker-compose.yml` to orchestrate the frontend, backend, and PostgreSQL database services.
- Updated `tsconfig.base.json` to be moved to the frontend folder for Docker compatibility.
- Enabled seamless local development and testing with Docker containers.

---

## **Changes Made**
- Added `Dockerfile` for the backend (Spring Boot) in `apps/backend/`.
- Added `Dockerfile` for the frontend (React/Next.js) in `apps/frontend/`.
- Created and configured `docker-compose.yml` to run all services (frontend, backend, and PostgreSQL).
- Moved `tsconfig.base.json` into the `frontend` folder to prevent build context issues.

---

## **Testing Instructions**
1. **Ensure Docker Desktop** is installed and running.
2. In the project root, run the following command to build the containers:
   ```bash
   docker-compose build
3. Start the containers:
   ```bash
   docker-compose up
4. Verify:
- Backend: Accessible at `http://localhost:8080` (if needed).
- Frontend: Accessible at `http://localhost:3000`.
- Database: PostgreSQL service running on port `5432`.

---

## **Additional Notes**
- Ensure that environment variables for PostgreSQL (username, password, and database) match the backend configuration.
- The backend container exited during initial testing but was resolved by ensuring Maven builds the backend properly with:
   ```bash
   mvn package

---

## **Logs:** 
   
[docker-compose up logs.txt](https://github.com/user-attachments/files/17531514/docker-compose.up.logs.txt)
[docker-compose build logs.txt](https://github.com/user-attachments/files/17531515/docker-compose.build.logs.txt)
